src/modules/psql_connector.py

```diff
@@ -20,19 +25,30 @@ def _connect(conf):
         logging.info('Successfully connected with local PSQL')
         return conn
 
-    def _select_single_execute(self, query):
-        cur = self.conn.cursor()
-        cur.execute(query)
-        values = cur.fetchall()
-        return values[0][0]
+    async def _create_pool(self):
+        pool = await create_pool(
+            database=self.conf["database"],
+            user=self.conf["user"],
+            password=self.conf["password"],
+            host=self.conf.get("host", "localhost"),
+            port=self.conf.get("port", 5432),
+            max_inactive_connection_lifetime=0.05,
+            timeout=0.05)
+        return pool
 
-    def select_single(self, query):
+    async def _select_single_execute(self, query):
+        value = await self.pool.fetch(query, timeout=0.05)
+        return value[0][0]
+
+    async def select_single(self, query):
         try:
-            return self._select_single_execute(query)
+            async with timeout(0.05):
+                return await self._select_single_execute(query)
         except InterfaceError as e:
             if "connection already closed" in str(e).lower():
-                self.conn = self.connect()
-                return self._select_single_execute(query)
+                self.conn = loop.run_until_complete(self._connect(conf))
```

Comment:
Should we recreate self.pool instead of self.conn?

Reply:
Yes I have recreate self.pool in this new commit